assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -0,0 +1,180 @@
+
+	const {
+		attributes,
+		setAttributes,
+		toggleAttachment,
+	} = props;
+
+	const {
+		attachmentClass,
+		postId,
+		postType,
+		title,
+		wrapperStyle,
+	} = attributes;
+
+	const fetchSelectedPost = () => {
+		isStillMounted = true;
+		if ( postId ) {
+			const currentFetchRequest = fetchRequest = apiFetch( {
+				path: `/wp/v2/${ postType }s/${ postId }`,
+			} ).then(
+				( post ) => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( post );
+						setFailedToFetch( false );
+					}
+				}
+			).catch(
+				() => {
+					if ( isStillMounted && fetchRequest === currentFetchRequest ) {
+						setSelectedPost( null );
+						setFailedToFetch( true );
+					}
+				}
+			);
+		}
+	};
+
+	useEffect( () => {
+		fetchSelectedPost();
+		return () => {
+			isStillMounted = false;
+		};
+	}, [] );
+
+	useEffect( () => {
+		fetchSelectedPost();
+	}, [ postId ] );
+
+	const removePost = () => {
+		setFailedToFetch( false );
+		setAttributes( { postId: null } );
+		setSelectedPost( null );
+	};
+
+	// Since the Page Attachment content is "out of it's borders" then clicking
+	// on the links might sometimes lose focus instead. Using refs for workaround.
+	const removeLinkRef = useRef( null );
+	const closeButtonRef = useRef( null );
+
+	useElementClickDetector( closeButtonRef, () => {
+		toggleAttachment( false );
+	} );
+
+	useElementClickDetector( removeLinkRef, null, true );
+
+	return (
+		<div className="attachment-container">
+			<div className="attachment-wrapper">
+				<div className="attachment-header">
+					<span
+						tabIndex="0"
+						className="amp-story-page-attachment-close-button"
+						role="button"
+						onKeyDown={ ( event ) => {
+							if ( ENTER === event.keyCode || SPACE === event.keyCode ) {
+								toggleAttachment( false );
+								event.stopPropagation();
+							}
+						} }
+						ref={ closeButtonRef }
+					/>
+					<RichText
+						value={ title }
+						tagName="div"
+						wrapperClassName="amp-story-page-attachment-title"
+						onChange={ ( value ) => setAttributes( { title: value } ) }
+						placeholder={ __( 'Write Title', 'amp' ) }
+						onClick={ ( event ) => event.stopPropagation() }
+					/>
+					{ postId && (
+						<Button
+							className="remove-attachment-post"
+							onClick={ ( event ) => {
+								event.stopPropagation();
+								removePost();
+							} }
+							isLink
+							isDestructive
+							ref={ removeLinkRef }
+						>
+							{ __( 'Remove Post', 'amp' ) }
+						</Button>
+					) }
+				</div>
+				<div className={ attachmentClass } style={ wrapperStyle }>
+					{ selectedPost && selectedPost.content && (
+						<RawHTML>
+							{ `<h2>${ selectedPost.title.rendered }</h2>${ selectedPost.content.rendered }` }
+						</RawHTML>
+					) }
+					{ ( ! postId || failedToFetch ) && (
+						<>
+							{ failedToFetch && (
+								<span>{ __( 'The selected post failed to load, please select a new post', 'amp' ) }</span>
```

Comment:
What if it was just an intermittent issue and the user doesn't want to select any other post? Is a "Try Again" button an option?